### PR TITLE
Add missing test functions to `test_parallel_coordinate` for matplotlib backend

### PR DESCRIPTION
**Pull request by MasahitoKumada**
## Motivation

Related to #3226

## Description of the changes

Add missing test functions in tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py

### CODE REVIEW

**Comment by toshihikoyanase** [comment]
@nzw0301 Could you review this PR, please?

**Comment by nzw0301** [comment]
Hi, I'm afraid the CIs are failing due to accessing undefined attributes of matplotlib's figure. I believe that other Matplotlib tests, for example, https://github.com/optuna/optuna/blob/master/tests/visualization_tests/matplotlib_tests/test_slice.py, are helpful to implement missing test cases. Please do not paste Plotly's tests because they are different.

**Comment by MasahitoKumada** [comment]
Thank you. I will try to fix it.

**Comment by nzw0301** [comment]
any update? 🤔

**Comment by nzw0301** [comment]
Let me rename the PR title for traceability and the forthcoming release note.

**Comment by nzw0301** [comment]
Can I review this PR?

**Review by nzw0301 — CHANGES_REQUESTED** [review]
I'll add more comments later. But let me leave a few suggestions.

File: tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py [hunk 1]
```diff
@@ -21,28 +26,73 @@ def test_plot_parallel_coordinate() -> None:
...
+    fig = figure.get_figure()
+    assert fig.axes[1].get_ylabel() == "Objective Value"
+    assert fig.axes[1].get_ylim() == (0.0, 2.0)
+    assert len(figure.findobj(LineCollection)) == 1
+    assert figure.findobj(LineCollection)[0].get_array().tolist()[:-1] == [0.0, 2.0, 1.0]
+    assert fig.axes[2].get_ylim() == (1.0, 2.5)
+    assert [
+        fig.axes[2].get_lines()[0].get_ydata()[0],
+        fig.axes[2].get_lines()[0].get_ydata()[-1],
+    ] == [1.0, 2.5]
```
Comment (nzw0301) [inline tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py:60]:
```suggestion
    axes = figure.get_figure().axes
    assert len(axes) == 2 + 1
    assert axes[1].get_ylabel() == "Objective Value"
    assert axes[1].get_ylim() == (0.0, 2.0)
    assert len(figure.findobj(LineCollection)) == 1
    assert figure.findobj(LineCollection)[0].get_array().tolist()[:-1] == [0.0, 2.0, 1.0]
    assert axes[2].get_ylim() == (1.0, 2.5)
    assert [
        axes[2].get_lines()[0].get_ydata()[0],
        axes[2].get_lines()[0].get_ydata()[-1],
    ] == [1.0, 2.5]
    xticklabels = axes[0].get_xticklabels()
    assert xticklabels[0].get_text() == "Objective Value"
    assert xticklabels[1].get_text() == "param_a"
```

File: tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py [hunk 2]
```diff
@@ -21,28 +26,73 @@ def test_plot_parallel_coordinate() -> None:
...
 
     # Test with a customized target value.
     with pytest.warns(UserWarning):
         figure = plot_parallel_coordinate(
             study, params=["param_a"], target=lambda t: t.params["param_b"]
         )
-    assert len(figure.get_lines()) == 0
+
+    assert len(list(figure.get_figure().axes)) == 2 + 1
+    fig = figure.get_figure()
```
Comment (nzw0301) [inline tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py:69]:
```suggestion
    axes = figure.get_figure().axes
    assert len(axes) == 2 + 1
```

File: tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py [hunk 3]
```diff
@@ -21,28 +26,73 @@ def test_plot_parallel_coordinate() -> None:
...
+    assert fig.axes[1].get_ylabel() == "Objective Value"
+    assert fig.axes[1].get_ylim() == (0.0, 2.0)
```
Comment (nzw0301) [inline tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py:71]:
```suggestion
    assert axes[1].get_ylabel() == "Objective Value"
    assert axes[1].get_ylim() == (0.0, 2.0)
```

File: tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py [hunk 4]
```diff
@@ -21,28 +26,73 @@ def test_plot_parallel_coordinate() -> None:
...
+    assert len(figure.findobj(LineCollection)) == 1
+    assert figure.findobj(LineCollection)[0].get_array().tolist()[:-1] == [2.0, 0.0, 1.0]
+    assert fig.axes[2].get_ylim() == (1.0, 2.5)
```
Comment (nzw0301) [inline tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py:74]:
```suggestion
    assert axes[2].get_ylim() == (1.0, 2.5)
```

File: tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py [hunk 5]
```diff
@@ -21,28 +26,73 @@ def test_plot_parallel_coordinate() -> None:
...
+    assert [
+        fig.axes[2].get_lines()[0].get_ydata()[0],
+        fig.axes[2].get_lines()[0].get_ydata()[-1],
```
Comment (nzw0301) [inline tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py:77]:
```suggestion
        axes[2].get_lines()[0].get_ydata()[0],
        axes[2].get_lines()[0].get_ydata()[-1],
```

File: tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py [hunk 6]
```diff
@@ -21,28 +26,73 @@ def test_plot_parallel_coordinate() -> None:
...
+    ] == [1.0, 2.5]
```
Comment (nzw0301) [inline tests/visualization_tests/matplotlib_tests/test_parallel_coordinate.py:78]:
```suggestion
    ] == [1.0, 2.5]
    xticklabels = axes[0].get_xticklabels()
    assert xticklabels[0].get_text() == "Objective Value"
    assert xticklabels[1].get_text() == "param_a"    
```

**Review by nzw0301 — CHANGES_REQUESTED** [review]

**Comment by nzw0301** [comment]
I'm afraid that I'm not sure that I'm an appropriate person as a reviewer because I've suggested too many changes so far. 

@MasahitoKumada Could you update the rest part yourself? If that is so hard, we are happy to merge this PR once, then @toshihikoyanase  or I will create a follow-up PR of this PR.

**Comment by nzw0301** [comment]
I'm afraid that we cannot proceed with this PR as the previous way, for example, https://github.com/optuna/optuna/pull/3266 due to the lack of our time. So let me close this PR, I'll send a PR based on this PR. Thank you for your understanding.

**Pull request closed without merging**